### DISCUSSION TITLE
System variables without delay

### DISCUSSION THREAD

**Discussion by grubmueller** [SukramJ/aiohomematic#667] [Ideas]
Hello,
first of all thanks for the great work! The new integration works very well for months now on my system.
There's only one open point so far:
I have an alarm system running on the CCU.
I already could connect the alarm system to the homeassistant alarm panel template.
As the alarm system state is read by the CCUs system variable, changes may be delayed by 30 seconds due to the poll time of the system variables.
Formerly I was experimenting with ioBroker, and there it was possible to immediately get a system variables content by pressing virtual channel 50 on the CCU. Speaking, the normal polling of 30 s was OK for most use cases, but when I wanted ioBroker to immediately visualize some changes based on CCU programs, I just added a button press 50 at the end of the CCU program. Unfortunately I do not know how this was implemented in ioBroker. Could you think of a way of implementing such a feature into hahomatic?
Regards, Michael

**Comment by Baxxy13** [comment]
Ich sehe hier grob 2 Möglichkeiten:

A: der "ioBroker-Style"
Es bräuchte einen, durch die Integration bereitgestellten, Dienst der entweder eine oder alle SysVars bei Ausführung aus der Zentrale ausliest und entsprechend in HA aktualisiert. Auslöser könnte dann ein beliebiges Ereignis sein, auch ein Tastendruck auf der Zentrale.

B: der "Zentralen-Push-Style"
Man könnte z.B. mittels Homematic-Script Daten direkt von der Zentrale in einen Homeassistant-Sensor pushen und entsprechend dann darauf in HA reagieren.
[Hier](https://homematic-forum.de/forum/viewtopic.php?f=19&t=75406&hilit=curl+post) hatte ich mich damit mal beschäftigt.

A wäre m.E. die bessere Option, bedeutet aber Arbeit für den Entwickler.
B Bedarf einiges an Skill um das einzurichten, und pushen ist immer als 2.Wahl anzusehen.

**Reply by SukramJ** [reply]
Mit gefällt Variante A.

**Reply by SukramJ** [reply]
Mit 1.19.8 gibt es einen Service `homematicip_local.fetch_system_variables`.

**Reply by grubmueller** [reply]
Super! Werde das gleich nach Verfügbarkeit testen.

**Reply by grubmueller** [reply]
Soeben getestet und funktioniert. Danke!
